File: 74-search-a-2d-matrix/74-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        
        int m = matrix.size();
        int n = matrix[0].size();
        int ansrow = -1;
        for(int i=0;i<m;i++)
        {
          ansrow = i;
          if(target<=matrix[i][n-1])
            break;
        }
        
        for(int i = 0; i<n;i++)
        {
            if(target == matrix[ansrow][i])
                return true;
        }
        
        return false;
    }
};

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    
    boolean binarySearch(int[] arr, int target)
    {
       
    
      int start = 0, end = arr.length-1;  
       while(start<=end)
       {
            int mid = start + (end - start)/2;
            if(arr[mid]==target)
                return true;
            if(arr[mid] < target)
                start = mid + 1;
            else 
                end = mid-1;
       }
        
        return false;
    
    }
    public boolean searchMatrix(int[][] matrix, int target) {
        
        
        for(int i=0;i<matrix.length;i++)
        {
            if(binarySearch(matrix[i],target))
                return true;
        }
        
        return false;
    }
}